core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/util/SignerUtils.java

```diff
@@ -13,33 +13,40 @@
  * permissions and limitations under the License.
  */
 
-package software.amazon.awssdk.http.auth.internal.util;
+package software.amazon.awssdk.http.auth.aws.util;
 
-import static software.amazon.awssdk.http.auth.internal.util.HttpChecksumUtils.hash;
+import static software.amazon.awssdk.http.auth.aws.util.HttpChecksumUtils.hash;
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.STREAMING_UNSIGNED_PAYLOAD_TRAILER;
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.UNSIGNED_PAYLOAD;
 
+import java.io.IOException;
+import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 import java.time.Instant;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
+import java.util.Optional;
 import javax.crypto.Mac;
 import javax.crypto.spec.SecretKeySpec;
-import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.ContentStreamProvider;
 import software.amazon.awssdk.http.SdkHttpRequest;
-import software.amazon.awssdk.http.auth.internal.checksums.ContentChecksum;
-import software.amazon.awssdk.http.auth.internal.checksums.SdkChecksum;
-import software.amazon.awssdk.http.auth.spi.SignRequest;
-import software.amazon.awssdk.http.auth.spi.SignerProperty;
+import software.amazon.awssdk.http.auth.aws.checksum.ContentChecksum;
+import software.amazon.awssdk.http.auth.aws.checksum.SdkChecksum;
+import software.amazon.awssdk.http.auth.aws.internal.util.FifoCache;
+import software.amazon.awssdk.http.auth.aws.internal.util.SignerKey;
+import software.amazon.awssdk.http.auth.aws.internal.util.SigningAlgorithm;
+import software.amazon.awssdk.http.auth.aws.signer.CredentialScope;
 import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
 import software.amazon.awssdk.utils.BinaryUtils;
 import software.amazon.awssdk.utils.Logger;
-import software.amazon.awssdk.utils.Validate;
 import software.amazon.awssdk.utils.http.SdkHttpUtils;
 
 /**
  * Utility methods to be used by various AWS Signer implementations.
- * This class is strictly internal and is subjected to change.
+ * This class is protected and subject to change.
  */
-@SdkInternalApi
+@SdkProtectedApi
```

Comment:
Any way to avoid this needing to be protected? Or can we reduce the number of things it contains to strictly what is needed to be protected?

Reply:
I guess we can make things that aren't used cross-module package-private?

Reply:
SignerUtils though seems used in the crt and event stream modules. One option to reduce the things, is being ok with some duplication - e.g., safeParseLong.

Reply:
+1 to above. What's the line, though? What shouldn't be copy?

Reply:
I'm not sure. Given the approach, I think we might be able to reduce (e.g., safeParseLong), but not sure about eliminating. But similar to comment on SignerConstant, if it can be SdkInternalApi for this PR, we can discuss in next PR?